src/main.rs

```diff
@@ -94,10 +94,48 @@ fn default_filename(filename: Option<String>) -> Result<PathBuf, clap::error::Er
     ))
 }
 
+fn assert_no_uncommitted_changes(path: &PathBuf) -> Result<(), clap::error::Error> {
+    // Extract the filename itself, as well as the directory from `path`.
+    assert!(path.is_file());
+    let filename_without_path = path.file_name().unwrap().to_str().unwrap();
+    let directory = path.parent().unwrap().to_str().unwrap();
+
+    let output = if cfg!(target_os = "windows") {
+        std::process::Command::new("cmd")
+            .args([
+                "/C",
+                format!("cd {}; git status --porcelain", directory).as_str(),
+            ])
+            .output()
+            .expect("Failed to run `git status`")
+    } else {
+        std::process::Command::new("sh")
+            .arg("-c")
+            .arg(format!("cd {}; git status --porcelain", directory).as_str())
+            .output()
+            .expect("Failed to run `git status`")
+    };
+
+    let git_status = String::from_utf8_lossy(&output.stdout);
+
+    // This implies that the spec we're targeting has no uncommitted changes,
+    // and so we're safe to proceed with rewrapping.
+    if !git_status.contains(&filename_without_path) {
```

Comment:
Hmm, can you explain why it would complain about those two specs in different folers, if my command is `cd`'ing into the right directory?

I like what you have anyways, but I want to see why this is broken in the current state.

Reply:
If I'm reading your code right, you're basically doing `git status --porcelain|grep basename`. But `git status --porcelain` always runs as if it's at the root of the repo (and `git status` without `--porcelain` returns things like `../spec1/index.bs`), so even if there's a file changed in a different directory, if the basenames match, this function would fail.

Reply:
Also, oops, you need to make the path absolute before my code will work. See https://doc.rust-lang.org/std/path/struct.Path.html#method.canonicalize

Reply:
Well it is doing `cd` right before that, so it was only running `git status` in the directory where the single unique `.bs` file is, or `source`.

Anyways, I've opted to use the adjusted command that you wrote up, with the exception of passing in `filename_without_path` instead of `path` at the end, since I think that's right.

Reply:
Hmm, things seemed to work without `canonicalize`, how do I know if I need it?

Reply:
I think `filename_without_path` is also right.